### PR TITLE
SEQNG-786: Better change handling on the session queue

### PR DESCRIPTION
**Pull request by cquiroz**
A bug in a previous refactoring produced that not all changes to the global state be reflected on the session queue, e.g. when going from one step to the next

This PR fixes this and adds some improvements to reduce the amount of refreshes

### CODE REVIEW

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/circuit/StatusAndLoadedSequencesFocus.scala [hunk 1]
```diff
@@ -22,34 +21,57 @@ final case class SequenceInSessionQueue(id:            Observation.Id,
...
-                                        targetName:    Option[TargetName],
+                                        targetName:    Option[String],
                                         runningStep:   Option[RunningStep],
                                         nextStepToRun: Option[Int])
 
 object SequenceInSessionQueue {
-  implicit val order: Order[SequenceInSessionQueue] = Order.by(_.id)
-  implicit val ordering: scala.math.Ordering[SequenceInSessionQueue] =
-    order.toOrdering
+  implicit val eq: Eq[SequenceInSessionQueue] =
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/circuit/StatusAndLoadedSequencesFocus.scala]: This is the core fix. `Order` wasn't consistent with a proper `Eq`

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/circuit/StatusAndLoadedSequencesFocus.scala [hunk 2]
```diff
@@ -22,34 +21,57 @@ final case class SequenceInSessionQueue(id:            Observation.Id,
                                         active:        Boolean,
                                         loaded:        Boolean,
                                         name:          String,
-                                        targetName:    Option[TargetName],
+                                        targetName:    Option[String],
```
Comment (jluhrs) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/circuit/StatusAndLoadedSequencesFocus.scala]: Why the change from `TargetName` to `String`?

Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/circuit/StatusAndLoadedSequencesFocus.scala]: it was trying to fix a bug but i'll revert it
